templates/consumer/java/src/main/java/service/consumer/application/Application.java

```diff
@@ -0,0 +1,54 @@
+package service.consumer.application;
+
+import com.alibaba.fastjson.JSONObject;
+import org.bouncycastle.crypto.CryptoException;
+
+import iservice.sdk.entity.options.TxOptions;
+import iservice.sdk.core.ServiceClient;
+
+import service.consumer.service.Client;
+import service.consumer.service.ConsumerListener;
+import service.consumer.types.InputWithHeader;
+import service.consumer.types.ServiceRequest;
+import service.consumer.{{service_name}}.CallbackImpl;
+
+import java.io.IOException;
+
+public class Application {
+
+  public Client client;
+
+  public Application(String keyAlgorithm, String nodeRPCAddr, String nodeGRPCAddr, String chain_id, String fee) throws Exception {
+    client = new Client(keyAlgorithm, nodeRPCAddr, nodeGRPCAddr, chain_id, fee);
+  }
+
+  public void invoke(String keyName, String password, String keyPath, String feeCap, String sender, String input) throws IOException, CryptoException {
+    String address = importKey(keyName, password, password, keyPath);
+    ServiceClient serviceClient = client.getServiceClient();
+
+    InputWithHeader inputWithHeader = JSONObject.parseObject(input, InputWithHeader.class);
+    ServiceRequest serviceRequest = new ServiceRequest(keyName, password, inputWithHeader.getHeader(), inputWithHeader.getBody());
+    String amount = feeCap.replaceAll("[^0-9]", "");
+    String denom = feeCap.replaceAll( "[^a-z]", "");
```

Comment:
Isn't there some function to split coin string to amount and denom? `denom` can also caontains `[0-9]`